$OpenBSD$

 - Fixes the order of CFLAGS and LDFLAGS.
 - Uses the compiler within $CC with a fallback to cc.

https://bitbucket.org/pypy/pypy/pull-request/153

--- rpython/translator/platform/openbsd.py.orig	Tue May 21 10:09:59 2013
+++ rpython/translator/platform/openbsd.py	Sun Jun  2 05:17:07 2013
@@ -5,11 +5,16 @@ import os
 from rpython.translator.platform.bsd import BSD
 
 class OpenBSD(BSD):
-    DEFAULT_CC = "cc"
+    if os.environ.get("CC") is None:
+        DEFAULT_CC = "cc"
+    else:
+        DEFAULT_CC = os.environ.get("CC")
+
     name = "openbsd"
 
-    link_flags = os.environ.get("LDFLAGS", '-pthread').split()
-    cflags = os.environ.get("CFLAGS", "-O3 -pthread -fomit-frame-pointer -D_BSD_SOURCE").split()
+    link_flags = os.environ.get("LDFLAGS", "").split() + ['-pthread']
+    cflags = ['-O3', '-pthread', '-fomit-frame-pointer', '-D_BSD_SOURCE'
+             ] + os.environ.get("CFLAGS", "").split()
 
     def _libs(self, libraries):
         libraries=set(libraries + ("intl", "iconv", "compat"))
